lib/koans/15_protocols.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Protocols do
+  use Koans
+
+  defprotocol School do
+    def enrol(name)
+  end
+
+  defimpl School, for: SecondarySchool do
+    def enrol(name) do
+      "#{name} enrolled at secondary school"
+    end
+  end
+
+  defimpl School, for: DanceSchool do
+    def enrol(name) do
+      "#{name} enrolled for dance"
+    end
+  end
+
+  defmodule EveningClass do end
+
+  @schools %{
+    :secondary => School.SecondarySchool,
+    :dance => School.DanceSchool,
+    :evening => EveningClass
+  }
+
+  koan "Commonality is the secret" do
+    school = Map.get(@schools, :dance)
+    assert school.enrol("Emily") == ___
+  end
+
+  koan "When there is a difference there is no way" do
+    assert_raise ___, fn ->
```

Comment:
I was also unsure whether to have this koan. It was to demonstrate what happens if you try to call a function in a place where you expect polymorphism to take place and the module does not implement the interface.

If there is benefit in showing what happens when something doesn't adhere to the expected Protocol we can keep the koan, if not then I am happy to remove this koan.